### PR TITLE
Register a shutdown hook at init that invokes shutdown-agents

### PR DESCRIPTION
**Pull request by gbuisson**

### CODE REVIEW

**Comment by saintx** [comment]
LGTM

File: src/ctia/shutdown.clj [hunk 1]
```diff
@@ -31,4 +37,4 @@
              (fn [_]
                (.addShutdownHook
                 (Runtime/getRuntime)
-                (Thread. shutdown-ctia!))))))))
+                (Thread. shutdown-ctia-and-agents!))))))))
```
Comment (saintx) [inline src/ctia/shutdown.clj]: This makes sense.

**Comment by polygloton** [comment]
I would add `(shutdown-agents)` to the end of `shutdown-ctia!`.  Since we will always want to do that, we might as well hard-code it (and not make it a hook).  Also, that guarantees it goes last, just incase some fool tries to use the pool in a shutdown hook.  :)

**Comment by gbuisson** [comment]
@polygloton I tried that at first, the problem is that it makes the JVM unusable for the next tests.

**Pull request merged**